### PR TITLE
Restore the layout trace viewer

### PR DESCRIPTION
**Pull request by shinglyu**
This patch restores the layout trace viewer removed in https://github.com/servo/servo/commit/e7510ab90c3482233ca8b188d748216901041970 , with some changes that makes it work with https://github.com/servo/servo/pull/13214

Sorry for the minified 3rd party CSS frameworks, it was from the original version and I haven't got time to replace them with hand-crafted CSS. 

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

- [ ] There are tests for these changes OR
- [x] These changes do not require tests because is a stand-alone debugging tool

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/13432)

### CODE REVIEW

**Comment by shinglyu** [comment]
I'll also try to update the wiki to document this after the patch merged.

File: etc/layout_viewer/README [hunk 1]
```diff
@@ -0,0 +1,15 @@
+To generate a layout trace:
+
+```
+./mach run -d -Z trace-layout <url>
+```
+
+or
+
+```
+./servo --trace-layout <url>
```
Comment (mbrubeck) [inline etc/layout_viewer/README]: This doesn't seem to work. Should it be removed? Or changed to use `-Z trace-layout`?

File: etc/layout_viewer/viewer.html [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                    };
+
+                    reader.readAsText(file);
+                    return false;
+                };
+
+/*
+                $.getJSON( "layout_trace.json", function(data) {
+                    new_data_loaded(data);
+                });*/
```
Comment (mbrubeck) [inline etc/layout_viewer/viewer.html]: Should this be removed?

File: etc/layout_viewer/css/bootstrap-treeview.min.css [hunk 3]
```diff
@@ -0,0 +1 @@
+.list-group-item{cursor:pointer}span.indent{margin-left:10px;margin-right:10px}span.icon{margin-right:5px}
```
Comment (mbrubeck) [inline etc/layout_viewer/css/bootstrap-treeview.min.css]: Could you please add `*.min.css` as an excluded filename for the `mach grep` command in `python/servo/devenv_commands.py`? (`*.min.js` is already excluded.)

**Review by mbrubeck — CHANGES_REQUESTED** [review]
Works great! r=mbrubeck with some minor fixes.

File: etc/layout_viewer/css/bootstrap-treeview.min.css (hunk 3 above)
Comment (shinglyu) [inline etc/layout_viewer/css/bootstrap-treeview.min.css]: Fixed!

**Comment by shinglyu** [comment]
Rebased to latest master to get rid of the CI failures.

**Pull request merged**